src/spec-configuration/containerCollectionsOCI.ts

```diff
@@ -227,28 +236,59 @@ export async function fetchOCIManifestIfExists(params: CommonParams, ref: OCIRef
 	}
 	const manifestUrl = `https://${ref.registry}/v2/${ref.path}/manifests/${reference}`;
 	output.write(`manifest url: ${manifestUrl}`, LogLevel.Trace);
-	const manifest = await getManifest(params, manifestUrl, ref);
+	const manifestContainer = await getManifest(params, manifestUrl, ref);
 
-	if (!manifest) {
+	if (!manifestContainer || !manifestContainer.manifestObj) {
 		return;
 	}
 
-	if (manifest?.config.mediaType !== DEVCONTAINER_MANIFEST_MEDIATYPE) {
-		output.write(`(!) Unexpected manifest media type: ${manifest?.config.mediaType}`, LogLevel.Error);
+	const { manifestObj } = manifestContainer;
+
+	if (manifestObj.config.mediaType !== DEVCONTAINER_MANIFEST_MEDIATYPE) {
+		output.write(`(!) Unexpected manifest media type: ${manifestObj.config.mediaType}`, LogLevel.Error);
 		return undefined;
 	}
 
-	return manifest;
+	return manifestContainer;
 }
 
-export async function getManifest(params: CommonParams, url: string, ref: OCIRef | OCICollectionRef, mimeType?: string): Promise<OCIManifest | undefined> {
-	return await getJsonWithMimeType(params, url, ref, mimeType || 'application/vnd.oci.image.manifest.v1+json');
+export async function getManifest(params: CommonParams, url: string, ref: OCIRef | OCICollectionRef, mimeType?: string): Promise<ManifestContainer | undefined> {
+	const res = await getJsonWithMimeType<OCIManifest>(params, url, ref, mimeType || 'application/vnd.oci.image.manifest.v1+json');
+	if (!res) {
+		return undefined;
+	}
+
+	const { body, headers } = res;
+	const manifestStringified = JSON.stringify(body);
+
+	// Per the specification:
+	// https://github.com/opencontainers/distribution-spec/blob/v1.0.1/spec.md#pulling-manifests
+	// The registry server SHOULD return the canonical content digest in a header, but it's not required to.
+	// That is useful to have, so if the server doesn't provide it, recalculate it outselves.
+	let contentDigest = headers['docker-content-digest'] || headers['Docker-Content-Digest'];
+	if (!contentDigest) {
+		contentDigest = `sha256:${crypto.createHash('sha256').update(manifestStringified).digest('hex')}`;
```

Comment:
Do we rely on this being the same digest as computed on the response body? If so, we should compute it from the original response body (the Buffer object, as opposed to the parsed and then stringified one).

Reply:
Yes this definitely should be the same digest that everyone else computes (and that we originally computed)

Comparing what this returns to what I see on GHCR, the results hashes are equal.  

<img width="2552" alt="image" src="https://user-images.githubusercontent.com/23246594/232148183-971af2ef-3b2b-4962-93b6-e62e4c70abbb.png">

This is also how I generate the manifest at publish time (https://github.com/devcontainers/cli/blob/main/src/spec-configuration/containerCollectionsOCIPush.ts#L408), so if any changes are made, it should be made in both spots.

Reply:
It will be more robust to compute the hash on a Buffer with the originally received bits. `manifestStringified` is derived from that like `JSON.stringify(JSON.parse(buffer.toString()))` and the hash function will itself first convert the string back to a Buffer. These conversions are not guaranteed to get you back the original bits.

E.g.: `JSON.stringify(JSON.parse('{ "p": "\\u0022" }')) === '{"p":"\\""}'` (whitespace and escaping changes)